### PR TITLE
Remove //pkg/api/legacyscheme from CCM and CM

### PR DESCRIPTION
**Pull request by cici37**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

This PR is part of the issue [here](https://github.com/kubernetes/cloud-provider/issues/29). See [KEP-removing-in-tree-providers](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/20190125-removing-in-tree-providers.md) for more details.
This PR is to remove dependency //pkg/api/legacyscheme from CCM and CM

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/cloud-controller-manager/app/options/options.go [hunk 1]
```diff
@@ -269,6 +269,5 @@ func createRecorder(kubeClient clientset.Interface, userAgent string) record.Eve
 	eventBroadcaster := record.NewBroadcaster()
 	eventBroadcaster.StartLogging(klog.Infof)
 	eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: kubeClient.CoreV1().Events("")})
-	// TODO: remove dependence on the legacyscheme
-	return eventBroadcaster.NewRecorder(legacyscheme.Scheme, v1.EventSource{Component: userAgent})
+	return eventBroadcaster.NewRecorder(runtime.NewScheme(), v1.EventSource{Component: userAgent})
```
Comment (andrewsykim) [inline cmd/cloud-controller-manager/app/options/options.go:272]: I'm not super familiar with the event broadcaster, but I think we need to pass in a `Scheme` that has the GVK for events installed.

I think using the client-go scheme would work (`k8s.io/client-go/kubernetes/scheme`)? @cheftako can you confirm^ ?

Comment (cheftako) [reply cmd/cloud-controller-manager/app/options/options.go:272]: We definitely need a scheme with the GVKs for any resources we use (which here is Events). I think client-go's scheme makes the most sense.

File: cmd/controller-manager/app/serve.go [hunk 2]
```diff
@@ -31,17 +33,23 @@ import (
 	"k8s.io/component-base/configz"
 	"k8s.io/component-base/metrics/legacyregistry"
 	_ "k8s.io/component-base/metrics/prometheus/workqueue" // for workqueue metric registration
-	"k8s.io/kubernetes/pkg/api/legacyscheme"
 )
 
+func getLegacyschema() runtime.NegotiatedSerializer {
```
Comment (cheftako) [inline cmd/controller-manager/app/serve.go:38]: I think we should have a different name. We're not getting a schema here we getting a codec/serializer. Also I think we're moving away from legacy here, which is a good thing. We also need to make sure we are using the same scheme here with all the GVKs.

Comment (cici37) [reply cmd/controller-manager/app/serve.go]: I am using scheme from client-go/kubernetes/scheme. Please have a look

**Comment by andrewsykim** [comment]
@cici37 can we open a separate PR for the codec change in cmd/controller-manager?

**Review by andrewsykim — COMMENTED** [review]
/approve
/lgtm

**Comment by cici37** [comment]
@andrewsykim @cheftako  Split the changes into two commits. Please have a look. Thanks

**Review by andrewsykim — COMMENTED** [review]
/approve
/lgtm

cmd/cloud-controller-manager changes lgtm!

**Pull request merged**